### DISCUSSION TITLE
How to migrate an existing agent to workflow chat?

### DISCUSSION THREAD

**Discussion by LeoQuote** [langgenius/dify#3259] [Help]
Recently I upgraded to v0.6.0, found some of the apps can be changed to workflow but some don't .

I really want to continue using the previous api key and share link while upgrading to the workflow engine.

**Comment by crazywoola** [comment]
I think the upgrade process means to create a new application in workflow type while deleting the old ones. In this case, we can insert old api key records into the database, or we simply do update the app_id in the database in a single transaction.

I prefer the first way

> we can insert old api key records into the database

What do you think.

**Reply by LeoQuote** [reply]
I think it would fulfill my request. It would be nice as long as api key stays the same.
